src/writeckpt.cpp

```diff
@@ -229,58 +228,62 @@ mtcp_writememoryareas(int fd)
      * Also, on SUSE 12, if this region was part of heap, the protected region
      * may have the label "[heap]".  So, we also save the memory region if it
      * has label "[heap]", "[stack]", or  "[stack:XXX]".
+     *
+     * EDIT: Not sure which kernel sections has ---p permissions. Let's save all
+     * regions with ---p permissions without discrimination.
+     *
      */
 
-    if (!((area.prot & PROT_READ) || (area.prot & PROT_WRITE)) &&
```

Comment:
@karya0 I was talking about this in my previous comment. You need this check to avoid checkpointing libraries' protected regions.